DocumentsFromSnapshotMigration/README.md

```diff
@@ -71,6 +71,7 @@ To see the default shard size, use the `--help` CLI option:
 | --target-aws-region               | The AWS region for the target cluster. Required if using SigV4 authentication                                                                            |
 | --target-aws-service-signing-name | The AWS service signing name (e.g. 'es' for Amazon OpenSearch Service, 'aoss' for Amazon OpenSearch Serverless). Required if using SigV4 authentication  |
 | --target-insecure                 | Flag to allow untrusted SSL certificates for target cluster                                                                                              |
+| --target-compression              | Flag to enable request compression for target cluster                                                                                                    |
```

Comment:
Or alternatively, what are we missing to commit to always supporting compressed responses?

Reply:
`all of our supported target cluster will support compressed requests`, I know this needs to be updated with limitations, but we were declaring support for 7.10 target clusters on the [readme](https://github.com/opensearch-project/opensearch-migrations?tab=readme-ov-file#supported-source-and-target-versions)

Reply:
Elasticsearch and OpenSearch have supported compression for a long time... I don't think we are ever going to support a pre-7.10 migration so I think this should be on by default.

Reply:
If you feel very strongly that we cannot move forward without a cli option, then I think we need to restructure how we pass these low level parameters around in our system, I'd suggest adding a new arguments JCommander object that we can keep adding onto that includes all these 'advanced settings'.

This would be a case where I would support static references to keep our construction/callstack patterns cleaner so we don't have to pipe these int/bool values all over

Reply:
Reading [AWS Docs](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/gzip.html#:~:text=You%20can%20compress%20HTTP%20requests,with%20similar%20OpenSearch%20settings%2C%20http_compression), support is by default off on ES 6.x and on thereafter. (So i'm less concerned about target cluster support. 

On the [OpenSource](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/network-settings/) side, it's by default OFF when HTTPS is on to mitigate security concerns.

Given it's configurable, i'd rather leave a cli option in. Does your thoughts on this being an "advanced option" change in light of this?

Reply:
Are we sure universally that request compression is always a net gain for the client and the server?  I suspect that there will be a number of workloads where it will NOT be.  Until we're sure, having a flag to toggle this seems like a prudent hedge.  Compression also makes debugging considerably more difficult (I usually turn it off in both directions when trying to debug)

Reply:
I think we need to restructure how we pass these low level parameters around in our system, I'd suggest adding a new arguments JCommander object that we can keep adding onto that includes all these 'advanced settings'.

This would be a case where I would support static references to keep our construction/callstack patterns cleaner so we don't have to pipe these int/bool values all over

Reply:
Refactored into a TargetAdvancedArgs object. @peternied, let me know your thoughts

Reply:
Thanks for the update there.  I like the direction you took this, and I've created a jira for some additional follow up that might make sense to iterate on if we find ourselves injecting new parameters https://opensearch.atlassian.net/browse/MIGRATIONS-1972